spec/features/user_signup_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'spec_helper'
+require 'helpers/user_helpers'
+
+feature 'A new user wants to sign up' do
+  scenario 'for the first time' do
+    visit '/'
+    expect(User.count).to eq 0
+    expect(page).to have_content 'Welcome to Chitter'
+    click_link 'Sign up'
+    expect(page).to have_content 'Please enter your details'
+    expect { sign_up }.to change(User, :count).by(1)
+    expect(page).to have_content 'Registration confirmed'
+    expect(User.first.email).to eq 'test@test.com'
+  end
+
+  scenario 'with an email that is already taken' do
+    sign_up
+    expect(User.count).to eq 1
+    expect { sign_up }.to change(User, :count).by(0)
+    expect(page).to have_content 'This email is already taken'
+  end
+
+  scenario 'with a username that is already taken' do
+    sign_up
+    expect(User.count).to eq 1
+    expect { sign_up('newperson@test.com',
+                      'test123,
+                      test123,
+                      John Jones
+                      test_user1')
+     }.to change(User, :count).by(0)
```

Comment:
`end` at 31, 5 is not aligned with `expect { sign_up('newperson@test.com',` at 26, 4